File: python/백준/구현/14719 빗물.py
import sys

input = sys.stdin.readline

H, W = map(int, input().split())
block = list(map(int, input().split()))

water = 0

for i in range(1, W-1):
    left_max = max(block[:i])
    right_max = max(block[i+1:])

    min_block = min(left_max, right_max)
    
    if min_block - block[i] > 0:
        water += min_block - block[i]

print(water) 

File: cpp/백준/구현/14719 빗물.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int H, W;
    cin >> H >> W;

    vector<int> block(W);
    for (int i = 0; i < W; ++i) cin >> block[i];

    int water = 0;

    for (int i = 1; i < W - 1; ++i) {
        int left_max = *max_element(block.begin(), block.begin() + i);
        int right_max = *max_element(block.begin() + i + 1, block.end());

        int min_block = min(left_max, right_max);

        if (min_block - block[i] > 0) {
            water += min_block - block[i];
        }
    }

    cout << water << "\n";
    return 0;
}
